src/python_testing/TC_ACE_1_5.py

```diff
@@ -0,0 +1,152 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_ACE_1_5(MatterBaseTest):
+
+    async def read_currentfabricindex_expected_success(self, th) -> int:
+        cluster = Clusters.Objects.OperationalCredentials
+        attribute = Clusters.OperationalCredentials.Attributes.CurrentFabricIndex
+        current_fabric_index = await self.read_single_attribute_check_success(dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+        return current_fabric_index
+
+    async def write_acl(self, acl, th):
+        # This returns an attribute status
+        result = await th.WriteAttribute(self.dut_node_id, [(0, Clusters.AccessControl.Attributes.Acl(acl))])
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+
+    async def read_descriptor_expect_success(self, th):
```

Comment:
Is there a reason to break all these into separate methods if all they do is a single await?

maybe in the main code instead of:

```python
self.print_step(7, "TH1 reads DUT Endpoint 0 Descriptor cluster DeviceTypeList attribute")
await self.read_descriptor_expect_success(self.th1)

self.print_step(8, "TH1 reads DUT Endpoint 0 Basic Information cluster VendorID attribute")
await self.read_basic_expect_unsupported_access(self.th1)

self.print_step(9, "TH2 reads DUT Endpoint 0 Descriptor cluster DeviceTypeList attribute")
await self.read_descriptor_expect_unsupported_access(self.th2)

self.print_step(10, "TH2 reads DUT Endpoint 0 Basic Information cluster VendorID attribute")
await self.read_basic_expect_success(self.th2)
```

we can inline like:

```python
self.print_step(7, "TH1 reads DUT Endpoint 0 Descriptor cluster DeviceTypeList attribute")
await self.read_single_attribute_check_success(
    dev_ctrl=th, endpoint=0,
    cluster=Clusters.Objects.Descriptor,
    attribute=Clusters.Descriptor.Attributes.DeviceTypeList)

self.print_step(8, "TH1 reads DUT Endpoint 0 Basic Information cluster VendorID attribute")
await self.read_single_attribute_check_success(
    dev_ctrl=th, endpoint=0, 
    cluster=Clusters.Objects.BasicInformation,
    attribute= Clusters.BasicInformation.Attributes.VendorID)

# ...
```

This seems to be easier to follow without needing to look up various methods.

Reply:
Got it! I did it, I think now is more readable and compact.